### PR TITLE
[Fix] Error in sales invoice and POS if customer group not defined in the customer

### PR DESCRIPTION
**Pull request by rohitwaghchaure**
**Issue**
Traceback (most recent call last): File "/home/frappe/benches/bench-2017-07-27/apps/frappe/frappe/app.py", line 56, in application response = frappe.handler.handle() File "/home/frappe/benches/bench-2017-07-27/apps/frappe/frappe/handler.py", line 21, in handle data = execute_cmd(cmd) File "/home/frappe/benches/bench-2017-07-27/apps/frappe/frappe/handler.py", line 52, in execute_cmd return frappe.call(method, **frappe.form_dict) File "/home/frappe/benches/bench-2017-07-27/apps/frappe/frappe/__init__.py", line 922, in call return fn(*args, **newargs) File "/home/frappe/benches/bench-2017-07-27/apps/frappe/frappe/desk/form/save.py", line 22, in savedocs doc.save() File "/home/frappe/benches/bench-2017-07-27/apps/frappe/frappe/model/document.py", line 230, in save return self._save(*args, **kwargs) File "/home/frappe/benches/bench-2017-07-27/apps/frappe/frappe/model/document.py", line 263, in _save self.run_before_save_methods() File "/home/frappe/benches/bench-2017-07-27/apps/frappe/frappe/model/document.py", line 772, in run_before_save_methods self.run_method("validate") File "/home/frappe/benches/bench-2017-07-27/apps/frappe/frappe/model/document.py", line 666, in run_method out = Document.hook(fn)(self, *args, **kwargs) File "/home/frappe/benches/bench-2017-07-27/apps/frappe/frappe/model/document.py", line 887, in composer return composed(self, method, *args, **kwargs) File "/home/frappe/benches/bench-2017-07-27/apps/frappe/frappe/model/document.py", line 870, in runner add_to_return_value(self, fn(self, *args, **kwargs)) File "/home/frappe/benches/bench-2017-07-27/apps/frappe/frappe/model/document.py", line 660, in fn = lambda self, *args, **kwargs: getattr(self, method)(*args, **kwargs) File "/home/frappe/benches/bench-2017-07-27/apps/erpnext/erpnext/accounts/doctype/sales_invoice/sales_invoice.py", line 56, in validate super(SalesInvoice, self).validate() File "/home/frappe/benches/bench-2017-07-27/apps/erpnext/erpnext/controllers/selling_controller.py", line 33, in validate super(SellingController, self).validate() File "/home/frappe/benches/bench-2017-07-27/apps/erpnext/erpnext/controllers/stock_controller.py", line 17, in validate super(StockController, self).validate() File "/home/frappe/benches/bench-2017-07-27/apps/erpnext/erpnext/controllers/accounts_controller.py", line 47, in validate self.validate_due_date() File "/home/frappe/benches/bench-2017-07-27/apps/erpnext/erpnext/controllers/accounts_controller.py", line 125, in validate_due_date validate_due_date(self.posting_date, self.due_date, "Customer", self.customer, self.company) File "/home/frappe/benches/bench-2017-07-27/apps/erpnext/erpnext/accounts/party.py", line 302, in validate_due_date default_due_date = get_due_date(posting_date, party_type, party, company) File "/home/frappe/benches/bench-2017-07-27/apps/erpnext/erpnext/accounts/party.py", line 267, in get_due_date credit_days_based_on, credit_days = get_credit_days(party_type, party, company) File "/home/frappe/benches/bench-2017-07-27/apps/erpnext/erpnext/accounts/party.py", line 287, in get_credit_days frappe.db.get_value("Customer Group", customer_group, ["credit_days_based_on", "credit_days"]) TypeError: 'NoneType' object is not iterable

Fixed https://github.com/frappe/erpnext/issues/10147

### CODE REVIEW

File: erpnext/accounts/doctype/sales_invoice/pos.py [hunk 1]
```diff
@@ -378,8 +379,8 @@ def add_customer(data):
 	customer_doc.customer_name = data.get('full_name') or data.get('customer')
 	customer_doc.customer_pos_id = data.get('customer_pos_id')
 	customer_doc.customer_type = 'Company'
-	customer_doc.customer_group = frappe.db.get_single_value('Selling Settings', 'customer_group')
-	customer_doc.territory = frappe.db.get_single_value('Selling Settings', 'territory')
+	customer_doc.customer_group = frappe.db.get_single_value('Selling Settings', 'customer_group') or _("Government")
```
Comment (nabinhait) [inline erpnext/accounts/doctype/sales_invoice/pos.py]: What if government does not exists?

I think we should force user to enter default customer group and territory in selling settings, even better if it is specified in POS Profile.

File: erpnext/accounts/party.py [hunk 2]
```diff
@@ -270,9 +270,13 @@ def get_due_date(posting_date, party_type, party, company):
 		elif credit_days_based_on == "Last Day of the Next Month":
 			due_date = (get_first_day(posting_date, 0, 2) + datetime.timedelta(-1)).strftime("%Y-%m-%d")
 
+	if not due_date:
+		due_date = today()
```
Comment (nabinhait) [inline erpnext/accounts/party.py]: Today is not realistic if it's not a POS kind of sale. I think it should be entered manually.

**Review by nabinhait — CHANGES_REQUESTED** [review]

**Comment by shanmukhasai6** [comment]
In this regard can we assign a default customer group, so that the new customers go into the default group

**Comment by Olawale1** [comment]
Hi,

Facing the same issue. Initially, invoices weren't getting synced from the POS at all and this error kept popping up. I then manually set the Customer Group in the Customer list. After this, the invoices got synced but the records are still in the Customer Queue!

Hope this will be rectified in the patch too

Thanks

**Pull request merged**